src/realm/sync/network/default_socket.cpp

```diff
@@ -78,26 +78,33 @@ class DefaultWebSocketImpl final : public WebSocketInterface, public Config {
                                            const std::string_view* body) override
     {
         ErrorCodes::Error error;
+        bool was_clean;
 
         if (ec == websocket::Error::bad_response_3xx_redirection ||
-            ec == websocket::Error::bad_response_301_moved_permanently) {
+            ec == websocket::Error::bad_response_301_moved_permanently ||
+            ec == websocket::Error::bad_response_308_permanent_redirect) {
             error = ErrorCodes::WebSocket_MovedPermanently;
+            was_clean = true;
```

Comment:
nit: you could just set `was_clean` to `true` at the top and then set it to False for internal server error